<a href="https://colab.research.google.com/github/IronStark007/Data-Analyst-Portfolio/blob/master/Exploratory%20Data%20Analysis%20-%20Terrorism/Data_Cleaning_Terrorism.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exploratory Data Analysis - Terrorism
* The task is to - Perform ‘Exploratory Data Analysis’ on dataset ‘Global Terrorism’
* And anwers the question - As a security/defense analyst, try to find out the hot zone of terrorism.
* Finally answer all - What all security issues and insights you can derive by EDA?
* Dataset:<a target="_blank" href="https://bit.ly/2TK5Xn5">Click Here.</a>

## Author - Ansari Mohammed Ali Nasim

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
#loading necessary library
import pandas as pd  #for loading the datasets
import numpy as np  #for numerical computations
import matplotlib.pyplot as plt  #for visualization
import seaborn as sns  #for visualization

In [3]:
# to ignore the warnings
import warnings
warnings.filterwarnings(action="ignore")

In [4]:
#loading the dataset
df = pd.read_csv("/content/drive/MyDrive/Global Terrorism - START data/globalterrorismdb_0718dist.csv", encoding='latin1')

#printing the first five rows of the dataset
df.head()

eventid  iyear  imonth  iday  ... INT_IDEO  INT_MISC INT_ANY  related
0  197000000001   1970       7     2  ...        0         0       0      NaN
1  197000000002   1970       0     0  ...        1         1       1      NaN
2  197001000001   1970       1     0  ...       -9         1       1      NaN
3  197001000002   1970       1     0  ...       -9         1       1      NaN
4  197001000003   1970       1     0  ...       -9         1       1      NaN

[5 rows x 135 columns]

So out dataframe is loaded..!!

It is a big dataset consisting of 135 columns. So let done some preprocessing before doing EDA of the data.
## Preprocessing

In [5]:
#checking info about the data
df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Columns: 135 entries, eventid to related
dtypes: float64(55), int64(22), object(58)
memory usage: 187.1+ MB


Some columns contains -9 as missing values therfore we have to change them to NA values

In [6]:
df=df.replace(to_replace=-9,value=np.nan)

In [7]:
#checking null values ratios
nulldf=df.isna().sum()/len(df)
nulldf=pd.DataFrame({"Colnames":df.columns,"Null values":nulldf.values})
nulldf=nulldf[nulldf['Null values']>=0.25]
nulldf=nulldf.transpose()
null_col=nulldf.loc['Colnames'].to_list()
print("Null values with more than 25% null values:\n",null_col)
print('Number of columns with more than 25% null values:',len(null_col))

Null values with more than 25% null values:
 ['approxdate', 'resolution', 'location', 'summary', 'alternative', 'alternative_txt', 'attacktype2', 'attacktype2_txt', 'attacktype3', 'attacktype3_txt', 'targtype2', 'targtype2_txt', 'targsubtype2', 'targsubtype2_txt', 'corp2', 'target2', 'natlty2', 'natlty2_txt', 'targtype3', 'targtype3_txt', 'targsubtype3', 'targsubtype3_txt', 'corp3', 'target3', 'natlty3', 'natlty3_txt', 'gsubname', 'gname2', 'gsubname2', 'gname3', 'gsubname3', 'motive', 'guncertain2', 'guncertain3', 'nperps', 'nperpcap', 'claimed', 'claimmode', 'claimmode_txt', 'claim2', 'claimmode2', 'claimmode2_txt', 'claim3', 'claimmode3', 'claimmode3_txt', 'compclaim', 'weaptype2', 'weaptype2_txt', 'weapsubtype2', 'weapsubtype2_txt', 'weaptype3', 'weaptype3_txt', 'weapsubtype3', 'weapsubtype3_txt', 'weaptype4', 'weaptype4_txt', 'weapsubtype4', 'weapsubtype4_txt', 'weapdetail', 'nkillus', 'nkillter', 'nwoundus', 'nwoundte', 'propextent', 'propextent_txt', 'propvalue', 'propcomment', 

Since all these 90 columns contains more than 30% null values we have to remove it

Also there are 19 columns which are not relevant so we will remove them too

In [8]:
#dropping all the null columns & not important columns
not_imp=['eventid', 'country', 'region', 'specificity' , 'crit1' , 'crit2', 'crit3', 'multiple', 'attacktype1', 'targtype1', 'targsubtype1', 'natlty1', 'weaptype1', 'weapsubtype1', 'dbsource', 'INT_LOG', 'INT_IDEO', 'INT_MISC', 'INT_ANY']
null_col=null_col+not_imp
df.drop(null_col,axis=1,inplace=True)

In [9]:
# printing dataset
df.head()

iyear  imonth  iday  extended  ... nkill nwound property ishostkid
0   1970       7     2         0  ...   1.0    0.0      0.0       0.0
1   1970       0     0         0  ...   0.0    0.0      0.0       1.0
2   1970       1     0         0  ...   1.0    0.0      0.0       0.0
3   1970       1     0         0  ...   NaN    NaN      1.0       0.0
4   1970       1     0         0  ...   NaN    NaN      1.0       0.0

[5 rows x 29 columns]

In [10]:
#checking the shape of the data
df.shape

(181691, 29)

In [11]:
#saving the data for analysis
df.to_csv('/content/drive/MyDrive/Global Terrorism - START data/Global_terrorism_cleaned.csv',index=False)

So the data is in a good shape and we are good to move on to the EDA in tableau

__For Analysis see tableau dashboard__